### PR TITLE
Add share-safe proof report artifacts

### LINKED ISSUE #189: [P1][Artifacts] Add share-safe report mode
## Problem

Some generated benchmark and compare artifacts contain machine-specific paths and verbose project details. Users need a safer mode for sharing benchmark evidence publicly.

## Why it matters

Benchmark results are more useful when users can share them without manually editing every local path or environment-specific detail.

## Expected behavior

Add a share-safe artifact mode for benchmark and compare outputs.

## Acceptance criteria

- Document which generated artifacts are intended for local use and which are intended for sharing.
- Add a utility or option that rewrites local absolute paths into stable placeholders.
- Preserve useful benchmark metrics and evidence fields.
- Add tests for path placeholder rewriting.
- Keep full local reports available by default.

## Files likely touched

- compare/report artifact generation
- benchmark docs
- tests for artifact formatting
- `CHANGELOG.md`

## Test plan

- `npm run typecheck`
- `npm run build`
- `npm run test:run`

## Non-goals

- Do not upload artifacts anywhere.
- Do not remove full local reports.
- Do not claim complete sanitization of every possible project detail.

### PR DESCRIPTION
**Pull request by mohanagy**
## Summary
- add companion `report.share-safe.json` artifacts for compare, review-compare, and runner-backed benchmark flows
- centralize share-safe path and text sanitization for artifact reports and harden redaction across traversal, UNC, protocol-relative URL, rooted `://`, and `file://` edge cases
- update docs/changelog and expand regression coverage for share-safe artifact behavior

Closes #189

## Testing
- npm run typecheck
- npm run build
- npm run test:run

## Summary by CodeRabbit

* **New Features**
  * Added `report.share-safe.json` output for compare, review-compare, and benchmark --exec runs: sanitized, placeholder-based paths for safe sharing while keeping full local `report.json`.
  * Added a share-safe sanitization utility to rewrite/redact paths and URL-like tokens for shareable reports.

* **Documentation**
  * Updated guides and tutorials to describe share-safe reporting behavior.

* **Tests**
  * Expanded unit tests to cover share-safe report generation and sanitization.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mohanagy/graphify-ts/pull/209?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/infrastructure/benchmark/runner.ts [hunk 1]
```diff
@@ -247,26 +250,46 @@ export async function runBenchmarkPrompt(options: RunBenchmarkPromptOptions): Pr
...
+      `${JSON.stringify(
+        {
+          ...localReport,
+          share_safe_report: true,
+          artifacts: {
+            ...Object.fromEntries(
+              Object.entries(localReportArtifacts).map(([key, path]) => [key, toShareSafeArtifactPath(path, shareSafeRoots)]),
+            ),
+            share_safe_report: toShareSafeArtifactPath(artifacts.share_safe_report, shareSafeRoots),
+          },
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply src/infrastructure/benchmark/runner.ts:291]: Checked this against the current code and the existing runner-backed benchmark regression in `tests/unit/benchmark.test.ts`. The temp-output case already round-trips through `toShareSafeArtifactPath()` and still serializes `<artifact-root>/...` in `report.share-safe.json`, so I left this path unchanged.

File: src/infrastructure/review-compare.ts [hunk 2]
```diff
@@ -128,6 +134,36 @@ function portablePath(path: string): string {
...
+}
+
+function rewriteShareSafeStderr(
+  stderr: Record<ReviewCompareMode, string | null>,
+  roots: ShareSafePathRoots,
+): Record<ReviewCompareMode, string | null> {
+  return {
+    verbose: stderr.verbose === null ? null : sanitizeShareSafeText(stderr.verbose, roots),
+    compact: stderr.compact === null ? null : sanitizeShareSafeText(stderr.compact, roots),
+  }
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply src/infrastructure/review-compare.ts:170]: Fixed in 21ec37f. `rewriteShareSafeStderr()` now redacts credential-shaped stderr values before share-safe path rewriting, and `tests/unit/review-compare.test.ts` now covers API-key and Bearer-token leakage cases.

**Comment by mohanagy** [comment]
Addressed the remaining review follow-ups in 21ec37f.

- native_agent compare now emits `report.share-safe.json` alongside `report.json`
- the share-safe native report sanitizes artifact/result paths, including `result_path`
- Windows-sensitive compare/review-compare assertions now avoid JSON-escaped path mismatches
